src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java

```diff
@@ -791,6 +791,10 @@ private void hideResponsesGiverRecipient() {
             String name = emailNameTable.get(response.recipient);
             FeedbackQuestionAttributes question = questions.get(response.feedbackQuestionId);
             FeedbackParticipantType participantType = question.recipientType;
+            if (participantType == FeedbackParticipantType.SELF) {
+                // recipient type for self-feedback is the same as the giver type
+                participantType = question.giverType;
```

Comment:
Here

Reply:
For self-feedback question, `recipientType ` is always `FeedbackParticipantType.SELF` and therefore we need to get the name of the participant from `giverType`.